dbus.c

```diff
@@ -770,11 +772,16 @@ luaA_dbus_connect_signal(lua_State *L)
     luaA_checkfunction(L, 2);
     signal_t *sig = signal_array_getbyid(&dbus_signals,
                                          a_strhash((const unsigned char *) name));
-    if(sig)
+    if(sig) {
         luaA_warn(L, "cannot add signal %s on D-Bus, already existing", name);
-    else
+        lua_pushnil(L);
+        lua_pushfstring(L, "cannot add signal %s on D-Bus, already existing", name);
+        return 2;
+    } else {
         signal_connect(&dbus_signals, name, luaA_object_ref(L, 2));
-    return 0;
+        lua_pushboolean(L, 1);
```

Comment:
Minor nitpick: We include stdbool.h, so you could just say `true` instead of `1` here, but that's not highly important.